File: frontend/src/types/epg.ts
export interface ProxyConfig {
  type: 'HTTP' | 'SOCKS5';
  server: string;
  port: number;
  username?: string;
  password?: string;
}

export interface EPGSource {
  id?: number;
  name: string;
  url: string;
  enabled: boolean;
  last_sync?: string;
}

export interface EPGProgram {
  id: number;
  channel_id: string;
  channel_name: string;
  title: string;
  start_time: string;
  end_time: string;
  description?: string;
  source_id: number;
}

export interface EPGProgramsResponse {
  data: EPGProgram[];
  total: number;
}

export interface FetchProgramsParams {
  page: number;
  page_size: number;
  channel_name?: string;
  start_time?: string;
  end_time?: string;
}

export interface EPGSource {
  id?: number;
  name: string;
  url: string;
  last_update: string | null;
  active: boolean;
  proxy?: ProxyConfig;
  sync_interval?: number;
}


export interface EPGChannel {
  id?: number;
  display_name: string;
  channel_id: string;
  language: string;
  category?: string;
  logo_url?: string;
}

File: backend/models/epg.py
from pydantic import BaseModel
from typing import Optional

class EPGSource(BaseModel):
    id: Optional[int] = None
    name: str
    url: str
    last_update: Optional[str] = None
    active: bool = True
    sync_interval: Optional[int] = 6
    default_language: Optional[str] = 'en'

class EPGProgram(BaseModel):
    id: Optional[int] = None
    channel: str
    title: str
    start_time: str
    end_time: str
    description: Optional[str] = None
    source_id: int

class EPGChannel(BaseModel):
    id: Optional[int] = None
    display_name: str
    language: str
    category: Optional[str] = None
    logo_url: Optional[str] = None
    local_logo_path: Optional[str] = None

class DefaultChannelLogo(BaseModel):
    id: Optional[int] = None
    channel_name: str
    logo_url: str
    priority: Optional[int] = 0  # 优先级，用于处理同名频道的情况